larrecodnn/ImagePatternAlgs/Tensorflow/Modules/EvaluateROIEff_module.cc

```diff
@@ -47,104 +56,433 @@ class nnet::EvaluateROIEff : public art::EDAnalyzer {
+        maxEtick = tdctick;
+      }
+      else if (tdctick - tdctick_previous != 1) {
+        signal_starttick.push_back(tdctick);
+        signal_endtick.push_back(tdctick_previous);
+        signal_energydeposits.push_back(totalE);
+        signal_energy_max.push_back(maxE);
+        signal_max_tdctick.push_back(maxEtick);
+        totalE = e_deposit;
+        maxE = e_deposit;
+        maxEtick = tdctick;
       }
-      h_energy[view]->Fill(totalE);
-      for (auto& wire : wires) {
-        if (wire->Channel() != ch1) continue;
-        const recob::Wire::RegionsOfInterest_t& signalROI = wire->SignalROI();
-        for (const auto& range : signalROI.get_ranges()) {
-          const auto& waveform = range.data();
+      else if (tdctick - tdctick_previous == 1){
+        totalE += e_deposit;
+        if (maxE < e_deposit) {
+          maxE = e_deposit;
+          maxEtick = tdctick;
+        }
+      }
+      
+      tdctick_previous = tdctick;
+    
+    } // loop over timeSlices timeSlice
+    
+    signal_endtick.push_back(tdctick_previous); // for last one
+    signal_energydeposits.push_back(totalE); // for last one
+    signal_energy_max.push_back(maxE); // for last one
+    signal_max_tdctick.push_back(maxEtick); // for last one
+    
+    if (signal_starttick.size() == 0 || (signal_endtick.size()==1 && signal_endtick.back()==-999)) continue;
+
+    /*
+    // scan signals
+    cout << "signal_starttick.size(): " << signal_starttick.size() << endl;
+    cout << "signal_endtick.size(): " << signal_endtick.size() << endl;
+    cout << "signal_energydeposits.size(): " << signal_energydeposits.size() << endl;
+    cout << "signal_energy_max.size(): " << signal_energy_max.size() << endl;
+    cout << "signal_max_tdctick.size(): " << signal_max_tdctick.size() << endl;
+    for (size_t s=0; s<signal_starttick.size(); s++) {
+      cout << "....signal: " <<  s << endl;
+      cout << "start: " << signal_starttick[s] << " ; end: " << signal_endtick[s] << endl;
+      cout << "energy deposits: " << signal_energydeposits[s] << endl;
+      cout << "energy maxE: " << signal_energy_max[s] << endl;
+      cout << "energy maxE tdctick: " << signal_max_tdctick[s] << endl;
+    }
+    cout << "first tdc: " << signal_starttick[0] << "; last tdc: " << signal_endtick.back() << endl;
+    */
+
+    for (auto & wire : wires) {
+      if (wire->Channel() != ch1) continue;
+
+      const recob::Wire::RegionsOfInterest_t& signalROI = wire->SignalROI();
+
+      std::vector<float> vecADC = wire->Signal(); // a zero-padded full length vector filled with RoI signal.
+      
+      // loop over signals:
+      // a) if signal s is not in any ROI (including the case no ROI), fill h_energy with signal_energy_max[s];
+      // b) if signal s is in a ROI, check following signals that are also in this ROI, then use the maximum of signal_energy_max to fill h_energy and h_energy_roi. After this, the loop will skip to the signal that is not in this ROI.
```

Comment:
These lines are getting pretty long.  Can you wrap that comment onto the next line(s)?